service/src/main/java/uk/gov/hmcts/reform/fpl/service/DirectionHelperService.java

```diff
@@ -167,10 +167,12 @@ private boolean responseExists(DirectionResponse response, Element<Direction> di
     public void addAssigneeDirectionKeyValuePairsToCaseData(String assignee,
                                                             List<Element<Direction>> directions,
                                                             CaseDetails caseDetails) {
-        if (assignee.equals("courtDirections")) {
-            caseDetails.getData().put(assignee.concat("Custom"), extractPartyResponse(assignee, directions));
-        } else {
-            caseDetails.getData().put(assignee, extractPartyResponse(assignee, directions));
+        if (!assignee.equals(ALL_PARTIES.getValue())) {
+            if (assignee.equals(COURT.getValue())) {
+                caseDetails.getData().put(assignee.concat("Custom"), extractPartyResponse(assignee, directions));
```

Comment:
Haven't had a look in the rest of the code base yet but this appears to be the only place you are taking advantage of the fact that you have a String instead of an enum. Every other time you are comparing you are having to check the value of the enum. This just seems like extra work unless there is something that I am missing.

Reply:
opened up a tech debt ticket to address https://tools.hmcts.net/jira/browse/FPLA-964